### ISSUE TITLE
Version missing from EnqueuedAsset types

### ISSUE THREAD

**Issue by benknight** [wp-graphql/wp-graphql#2297]
Script and stylesheet versions don't resolve correctly because they reference an undefined filed named `$version` on the dependency objects. Desired behavior for the resolver would be to return `$ver` if it's set, and fall back to `$default_version` so that consumers can use query string `?v=...` to manage caching the same way native themes do.

https://github.com/wp-graphql/wp-graphql/blob/develop/src/Type/InterfaceType/EnqueuedAsset.php#L53

See:

https://developer.wordpress.org/reference/classes/wp_dependencies/
https://developer.wordpress.org/reference/classes/_wp_dependency/
https://developer.wordpress.org/reference/classes/wp_scripts/

**Comment by justlevine** [comment]
Hey @benknight thanks for reporting the issue, and the possible fix!

Few notes after investigation:
- Many scripts and stylesheets dont have a `$ver` specified in `global $wp_{scripts|styles}`, which as far as I can tell is on only programmatic way to get the list of assets. Please correct me if I'm wrong.
- The `$default_version` doesnt exist on the individual `_WP_Dependency`, but instead is a property of WP_{Scripts|Styles}. We can get it by getting both globals and then using the `$type` to get the property dynamically.

> consumers can use query string `?v=...` to manage caching the same way native themes do.

Not 100% I understand what this would look like in headless. My experience with managing scripts/styles in WP is mainly configuring caching plugins (in native), or stripping them out entirely (aka why I ♥ decoupled ). So while this seems like an easy PR, it's probably best someone who understands the nuances writes it.

### Code sample
(Assuming there arent any special cases to consider)
```php
// EnqueuedAsset.php::L53
'version' => [
  'type'        => 'String',
  'description' => __( 'The version of the enqueued asset', 'wp-graphql' ),
  'resolve' => static function( $asset ) {
     global $wp_styles, $wp_scripts;
   
     $version = ! empty( $asset->ver ) ? $asset->ver : ${'wp_' . $asset->type . 's'}->default_version;
     return $version ?: null;
  },
]
```

**Comment by benknight** [comment]
> Not 100% I understand what this would look like in headless. My experience with managing scripts/styles in WP is mainly configuring caching plugins (in native), or stripping them out entirely (aka why I ♥ decoupled ). So while this seems like an easy PR, it's probably best someone who understands the nuances writes it.

@justlevine In my case, we encountered a bug because in order to use blocks, it's necessary to include the blocks stylesheet at `/wp-includes/css/dist/block-library/style.min.css`. However in a recent version of WordPress many of the block's styles changed which broke our post styles because many users still had the old stylesheet cached. This is solved in native by appending the version number to the stylesheet URL e.g. `?v=5.1.2`. We also manually include some plugin stylesheets/scripts for things like extended block libraries like Stackable for instance. These are vulnerable to the same caching issues if the authors were to push an update. So in other words, we need to be able to replicate native's cache-busting behavior using version numbers which should follow the same logic of using the dependency's version if specified, while following back to the default version (which is just the WordPress semver string).

**Comment by justlevine** [comment]
#2259 is tracking that, and you can use the provided snippet locally in the interim - only thing that isnt are layout styles, since theyre not saved in `global-styles` (see WordPress/gutenberg/issues/35376). 

Regarding this issue, i'm going to give it a few days in case someone shares an edge case for the version that needs to be included, and then if nobody else has yet will get to work on the PR.

**Comment by justlevine** [comment]
merged in v1.8.3
